examples/client_forge/client_forge.js

```diff
@@ -0,0 +1,191 @@
+var mc = require('minecraft-protocol');
+var ProtoDef = require('protodef').ProtoDef;
+
+if(process.argv.length < 4 || process.argv.length > 6) {
+  console.log("Usage : node echo.js <host> <port> [<name>] [<password>]");
+  process.exit(1);
+}
+
+var client = mc.createClient({
+  forge: true,
+  host: process.argv[2],
+  port: parseInt(process.argv[3]),
+  username: process.argv[4] ? process.argv[4] : "echo",
+  password: process.argv[5]
+});
+
+client.on('connect', function() {
+  console.info('connected');
+});
+client.on('disconnect', function(packet) {
+  console.log('disconnected: '+ packet.reason);
+});
+client.on('chat', function(packet) {
+  var jsonMsg = JSON.parse(packet.message);
+  if(jsonMsg.translate == 'chat.type.announcement' || jsonMsg.translate == 'chat.type.text') {
+    var username = jsonMsg.with[0].text;
+    var msg = jsonMsg.with[1];
+    if(username === client.username) return;
+    client.write('chat', {message: msg});
+  }
+});
+
+var proto = new ProtoDef();
+// copied from ../../dist/transforms/serializer.js TODO: refactor
+proto.addType("string", ["pstring", {
+      countType: "varint"
+    }]);
+
+
+// http://wiki.vg/Minecraft_Forge_Handshake
+proto.addType('FML|HS',
+  [
+    'container',
+    [
+      {
+        "name": "discriminator",
+        "type": "byte"
+      },
+
+      // ServerHello
```

Comment:
(as I was saying in gitter), it might make more sense to use a big switch+ a mapper, like in https://github.com/PrismarineJS/prismarine-nbt/blob/master/nbt.json#L48 (or https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/transforms/serializer.js#L21 )
* mapper makes it possible for user to use the names (for example ServerHello instead of 0)
* the big switch probably makes more sense because fields are distinct between each kind of "packet" here